update.ps1

```diff
@@ -65,145 +50,191 @@ function Get-ADPAccessToken {
+        }
+        $responseGetUser = Invoke-RestMethod @splatParams
 
-    if ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -ne $account.workerEmail) {
+    if ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -ne $actionContext.Data.workerEmail) {
         $action = 'Update'
-        $msg = "$action ADPWorkforce E-mail address: [$($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri)] to [$($account.workerEmail)] for: [$($p.DisplayName)] will be executed during enforcement"
+        $msg = "$action ADPWorkforce E-mail address: [$($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri)] to [$($actionContext.Data.workerEmail)] for: [$($personContext.Person.DisplayName)] will be executed during enforcement"
     }
-    elseif ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -eq $account.workerEmail) {
+    elseif ($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri -eq $actionContext.Data.workerEmail) {
         $action = 'Exit'
-        $msg = "E-mail address: [$($account.workerEmail)] for: [$($p.DisplayName)] does not require an update"
+        $msg = "E-mail address: [$($actionContext.Data.workerEmail)] for: [$($personContext.Person.DisplayName)] does not require an update"
     }
 
-    switch ($action) {
-        'Update' {
-            Write-Verbose "Updating ADPWorkforce account: [$($aRef)] for: [$($p.DisplayName)]"
-            $body = @{
-                events = @(@{
-                        eventNameCode = @{
-                            codeValue = 'worker.businessCommunication.email.change'
-                        }
-                        data          = @{
-                            eventContext = @{
-                                worker = @{
-                                    workerID = @{
-                                        idValue = $account.workerId
+
+    $correlatedAccount = $responseGetUser
+    $outputContext.PreviousData = $responseGetUser
+
+    # Always compare the account against the current account in target system
+    if ($null -ne $correlatedAccount) {
+        $splatCompareProperties = @{
+            ReferenceObject  = $correlatedAccount.PSObject.Properties
+            DifferenceObject = $actionContext.Data.PSObject.Properties
+        }
+        $propertiesChanged = Compare-Object @splatCompareProperties -PassThru | Where-Object { $_.SideIndicator -eq '=>' }
+        if ($propertiesChanged) {
+            $action = 'UpdateAccount'
+            $dryRunMessage = "Account property(s) required to update: $($propertiesChanged.Name -join ', ')"
+        } else {
+            $action = 'NoChanges'
+            $dryRunMessage = 'No changes will be made to the account during enforcement'
+        }
+    } else {
+        $action = 'NotFound'
+        $dryRunMessage = "ADPWorkforce account for: [$($personContext.Person.DisplayName)] not found. Possibly deleted."
+    }
```

Comment:
I think we can remove this compare logic since we already do a compare on lines 115/122.